Source/ASRunLoopQueue.mm

```diff
@@ -196,8 +200,58 @@ @interface ASRunLoopQueue () {
 
 @end
 
+#if AS_KDEBUG_ENABLE
+typedef enum {
+  kCATransactionPhasePreLayout,
+  kCATransactionPhasePreCommit,
+  kCATransactionPhasePostCommit,
+} CATransactionPhase;
+
+@interface CATransaction (Private)
++ (void)addCommitHandler:(void(^)(void))block forPhase:(CATransactionPhase)phase;
++ (int)currentState;
+@end
+#endif
+
 @implementation ASRunLoopQueue
 
+#if AS_KDEBUG_ENABLE
++ (void)load
+{
+  [self registerCATransactionObservers];
+}
+
++ (void)registerCATransactionObservers
+{
+  static dispatch_block_t preLayoutHandler;
+  static dispatch_block_t preCommitHandler;
+  static dispatch_block_t postCommitHandler;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    preLayoutHandler = ^{
```

Comment:
I like the look of this but you can't put breakpoints inside macros unfortunately, so I kept the ugly code.